src/Microsoft.ML.Data/DataLoadSave/FakeSchema.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.ML.Data.DataLoadSave
+{
+
+    /// <summary>
+    /// A fake schema that is manufactured out of a SchemaShape.
+    /// It will pretend that all vector sizes are equal to 10, all key value counts are equal to 10,
+    /// and all values are defaults (for metadata).
+    /// </summary>
+    internal sealed class FakeSchema : ISchema
+    {
+        private readonly IHostEnvironment _env;
+        private readonly SchemaShape _shape;
+        private readonly Dictionary<string, int> _colMap;
+
+        public FakeSchema(IHostEnvironment env, SchemaShape inputShape)
+        {
+            _env = env;
+            _shape = inputShape;
+            _colMap = Enumerable.Range(0, _shape.Columns.Length)
+                .ToDictionary(idx => _shape.Columns[idx].Name, idx => idx);
+        }
+
+        public int ColumnCount => _shape.Columns.Length;
+
+        public string GetColumnName(int col)
+        {
+            _env.Check(0 <= col && col < ColumnCount);
+            return _shape.Columns[col].Name;
+        }
+
+        public ColumnType GetColumnType(int col)
+        {
+            _env.Check(0 <= col && col < ColumnCount);
+            var inputCol = _shape.Columns[col];
+            return MakeColumnType(inputCol);
+        }
+
+        public bool TryGetColumnIndex(string name, out int col) => _colMap.TryGetValue(name, out col);
+
+        private static ColumnType MakeColumnType(SchemaShape.Column inputCol)
+        {
+            ColumnType curType = inputCol.ItemType;
+            if (inputCol.IsKey)
+                curType = new KeyType(curType.AsPrimitive.RawKind, 0, 10);
```

Comment:
>10 [](start = 70, length = 2)

I see it being used at multiple places. Can we define a `const` for it?

Reply:
Sure

---
In reply to: [218248484](https://github.com/dotnet/machinelearning/pull/931#discussion_r218248484) [](ancestors = 218248484)